implementation/rust-embedded-library.tex

```diff
@@ -35,18 +21,19 @@ \subsection{The Core Library}
 \end{table}
 
 The memory management functions given in \autoref{tab:core:definitions} are provided by \lib{newlib} and are exposed through the \lib{startup} library described in \autoref{sec:startup}.
+These functions provide the basic memory management that is needed in order to utilize the parts of \gls{rsl} that defines dynamically allocated data structures.
+
+\concept{Panicking} is {\rust}'s way of unwinding the currently executing thread, ultimately resulting in the thread being terminated.
+A panic in {\rust} can happen when e.g. an array is indexed out of bounds, which causes the \code{rust\_begin\_unwind} function to be called.
```

Comment:
I think this particular use of `e.g.` is wrong. 

This might be better: `A panic in {\rust} can happen, e.g. an array is indexed out of bounds, which causes the \code{rust\_begin\_unwind} function to be called.`